README.md

```diff
@@ -85,24 +99,105 @@ let client = OpenFeatureAPI.shared.getClient()
+If the hook you're looking for hasn't been created yet, see the [develop a hook](#develop-a-hook) section to learn how to build it yourself.
 
-Please refer to our [documentation on static-context APIs](https://github.com/open-feature/spec/pull/171) for further information on how these APIs are structured for the use-case of mobile clients.
+Once you've added a hook as a dependency, it can be registered at the global, client, or flag invocation level.
 
-### Events
+```swift
+// add a hook globally, to run on all evaluations
+OpenFeatureAPI.shared.addHooks(hooks: ExampleHook())
+
+// add a hook on this client, to run on all evaluations made by this client
+val client = OpenFeatureAPI.shared.getClient()
+client.addHooks(ExampleHook())
+
+// add a hook for this evaluation only
+_ = client.getValue(
+    key: "key",
+    defaultValue: false,
+    options: FlagEvaluationOptions(hooks: [ExampleHook()]))
+```
+
+### Eventing
 
 Events allow you to react to state changes in the provider or underlying flag management system, such as flag definition changes, provider readiness, or error conditions.
-Initialization events (`PROVIDER_READY` on success, `PROVIDER_ERROR` on failure) are emitted for every provider.
+Initialization events (`PROVIDER_READY` on success, `PROVIDER_ERROR` on failure) are dispatched for every provider.
 Some providers support additional events, such as `PROVIDER_CONFIGURATION_CHANGED`.
-Please refer to the documentation of the provider you're using to see what events are supported and when they are emitted.
 
-To register a handler for API level provider events, use the `OpenFeatureAPI` `addHandler(observer:selector:event:)` function. Event handlers can also be removed using the equivalent `removeHandler` function. `Client`s also provide add and remove functions for listening to that specific client. A `ProviderEvent` enum is defined to ease subscription.
+Please refer to the documentation of the provider you're using to see what events are supported.
 
-Events can contain extra information in the `userInfo` dictionary of the notification. All events will contain a reference to the provider that emitted the event (under the `providerEventDetailsKeyProvider` key). Error events will also provide a reference to the underlying error (under the `providerEventDetailsKeyError` key). Finally, client specific events will contain a reference to the client (under the `providerEventDetailsKeyClient` key)
+```swift
+OpenFeatureAPI.shared.addHandler(
+    observer: self, selector: #selector(readyEventEmitted(notification:)), event: .ready
+)
 
-### Providers
+func readyEventEmitted(notification: NSNotification) {
+    // to something now that the provider is ready
+}
+```
+
+### Shutdown
+
+The OpenFeature API provides a close function to perform a cleanup of all registered providers.
+This should only be called when your application is in the process of shutting down.
+
+<!-- TODO: code example for global shutdown -->
```

Comment:
Please add a code example.

Reply:
On second look, there is no explicit shutdown functionality yet. I marked this as "not implemented" ❌